### PR TITLE
Datasource v2

### PR DESCRIPTION
**Pull request by davidrabinowitz**
First working version, before code clean ups. Still at POC level!

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -4,7 +4,7 @@ lazy val sparkVersion = "2.4.0"
 
 lazy val commonSettings = Seq(
   organization := "com.google.cloud.spark",
-  version := "0.14.1-beta-SNAPSHOT",
+  version := "0.15.2-beta-SNAPSHOT",
```
Comment (Gaurangi94) [inline build.sbt:7]: why is it 0.15.2-beta-SNAPSHOT here when it 0.15.1 in Readme?

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientModule.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.Optional;
+
+public class BigQueryClientModule implements Module {
+
+    @Override
+    public void configure(Binder binder) {
+        // BigQuery related
+        binder.bind(BigQueryStorageClientFactory.class).in(Scopes.SINGLETON);
+
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientModule.java:18]: nit: Remove extra new line?

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryStorageClientFactory.java [hunk 3]
```diff
@@ -16,39 +16,43 @@
...
 import com.google.auth.Credentials;
 import com.google.cloud.bigquery.storage.v1beta1.BigQueryStorageClient;
 import com.google.cloud.bigquery.storage.v1beta1.BigQueryStorageSettings;
 
 import javax.inject.Inject;
 import java.io.IOException;
+import java.io.Serializable;
 import java.io.UncheckedIOException;
-import java.util.Optional;
+import com.google.common.base.Optional;
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryStorageClientFactory.java:29]: should we move this import to line 24 to keep all the imports sorted?

File: connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceV2.java [hunk 4]
```diff
@@ -15,21 +15,46 @@
...
+import org.apache.spark.sql.SparkSession;
 import org.apache.spark.sql.sources.v2.DataSourceOptions;
 import org.apache.spark.sql.sources.v2.DataSourceV2;
 import org.apache.spark.sql.sources.v2.ReadSupport;
 import org.apache.spark.sql.sources.v2.reader.DataSourceReader;
 import org.apache.spark.sql.types.StructType;
 
+import java.util.Optional;
+
+import static scala.collection.JavaConversions.mapAsJavaMap;
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceV2.java:31]: sort imports?

File: build.sbt (hunk 1 above)
Comment (davidrabinowitz) [reply build.sbt:7]: The README references the last released version which is 0.15.1-beta. the build.sbt has the next release version.

File: connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceV2.java (hunk 4 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceV2.java:31]: They are, based on intellij - static import at the end, java runtime after libraries. We will apply the [Google java code formatter](https://github.com/google/google-java-format) when everything is merged to master.

**Pull request merged**